zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -150,3 +233,60 @@ def __init__(self, *args, **kwargs):
             }
         }
     }
+
+
+class HikingPowerMeter(TuyaSwitch):
+    """Hiking Power Meter Device - DDS238-2."""
+
+    signature = {
+        # "node_descriptor": "<NodeDescriptor byte1=1 byte2=64 mac_capability_flags=142 manufacturer_code=4098
+        #                       maximum_buffer_size=82 maximum_incoming_transfer_size=82 server_mask=11264
+        #                       maximum_outgoing_transfer_size=82 descriptor_capability_field=0>",
+        # device_version=1
+        # input_clusters=[0x0000, 0x0004, 0x0005, 0xef00]
+        # output_clusters=[0x000a, 0x0019]
+        MODELS_INFO: [("_TZE200_bkkmqmyo", "TS0601")],
+        ENDPOINTS: {
+            # <SimpleDescriptor endpoint=1 profile=260 device_type=51
+            # device_version=1
+            # input_clusters=[0, 4, 5, 61184]
+            # output_clusters=[10, 25]>
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    TuyaManufClusterAttributes.cluster_id,
+                ],
+                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
+            }
+        },
+    }
+
+    replacement = {
+        ENDPOINTS: {
+            1: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
+                INPUT_CLUSTERS: [
+                    Basic.cluster_id,
+                    Groups.cluster_id,
+                    Scenes.cluster_id,
+                    HikingManufClusterDinPower,
+                    TuyaElectricalMeasurement,
+                    TuyaPowerMeasurement,
+                ],
+                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
+            },
+            16: {
+                PROFILE_ID: zha.PROFILE_ID,
+                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
+                INPUT_CLUSTERS: [
+                    TuyaOnOff,
+                ],
+                OUTPUT_CLUSTERS: [],
+            },
```

Comment:
Why setting the `TuyaOnOff` cluster in another endpoint? Since the `DEVICE_TYPE` is the same, couldn't it go to endpoint 1?

Reply:
Originally it did live under endpoint 1, but the switch did not toggle. I looked into the `TuyaOnOff` class and found the following code:
```python
            cmd_payload.command_id = TUYA_CMD_BASE + self.endpoint.endpoint_id
```
https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/tuya/__init__.py#L519

`TUYA_CMD_BASE` has the value of `0x0100`
https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/tuya/__init__.py#L110

My device has the On/Off Switch accessible under attribute: `0x0110`
So the only way to get  `cmd_payload.command_id` to be equal to my switch's attribute was put it under endpoint 16 (`0x0010`)

Reply:
🤦🏻‍♂️ That makes me wonder why I didn't migrate the quirk to the MCU classes at the time...

Well, the implementation can be left as is, but you could also tune it here:
```python
         if attrid == HIKING_DIN_SWITCH_ATTR:
             self.endpoint.device.switch_bus.listener_event(SWITCH_EVENT, 0x0101, value)
```

Reply:
Is there anything that I need to do to get this PR merged?